### DISCUSSION TITLE
with IRealObject: support using record instead of class for db entities.

### DISCUSSION THREAD

**Discussion by softlion** [realm/realm-dotnet#3303] [Ideas]
a `record` is a class that supports a very compact declaration.

So instead of writing:

```c#
public partial class DbWallet : IRealmObject 
{  
    public string Id {get;set;}
    //...
}
```

it would be cool that it supports writing:

```c#
public partial record DbWallet(string Id, ...) : IRealmObject;
```

Much less boiler plate code to write.  
Easy to support (make sure it's not a record struct !).

**Comment by nirinchev** [comment]
@papafe this sounds interesting (and doable) - do you have opinions on whether we can do it?

**Reply by papafe** [reply]
I think this is something that we could add. Overall the generation behaviour should be the same, we mostly need to improve our pipeline to include the positional properties too in the generation. 
We probably also need to take a look at the equality operators that are synthesised for records, so that they behave the same as for "normal" classes.

**Reply by softlion** [reply]
You are right, the equality operators on record have some specificities, that is something you need to check.

**Reply by papafe** [reply]
@softlion definitely. 
By the way, I understand the class declaration would look much more compact, but the positional properties can only be set during initialisation, so I have a feeling the applicability of record classes with Realm objects is somehow limited. It is possible to change this by specifying the property manually, but in that case the definition will be exactly the same as with a normal class, and there is no advantage in compactness. 
@softlion Are you thinking about some specific use case maybe?
